sonar-db/src/main/java/org/sonar/db/component/SnapshotDao.java

```diff
@@ -69,6 +69,12 @@ public boolean hasLastSnapshotByComponentUuid(DbSession session, String componen
     return mapper(session).selectPreviousVersionSnapshots(componentId, lastVersion);
   }
 
+  @CheckForNull
+  public SnapshotDto selectFirstSnapshot(DbSession session, long componentId) {
+    List<SnapshotDto> snapshotDtos = mapper(session).selectSnapshotsOrderedByAscCreatedAt(componentId);
```

Comment:
Filtering the first row must be done in DB to avoid loading of huge history of snapshots. See RowBounds of MyBatis.